### ISSUE TITLE
Type lost when calling asyncIterator.call

### ISSUE THREAD

**Issue by wong2** [Sec-ant/readable-stream#29]
I have the following code:
```
for await (const event of asyncIterator.call(eventStream)) {
  ...
}
```
Where `eventStream` is of type `ReadableStream<ParsedEvent>`, but the `event` is now `unknown`.

**Comment by Sec-ant** [comment]
Thanks for reporting this issue. It's a [TypeScript limitation](https://github.com/microsoft/TypeScript/issues/54707) that generics are lost in `fn.call()`.

I refactored the `asyncIterator` ponyfill API in #30 to make it more functional.

Can you test it by installing the [preview release package](https://github.com/Sec-ant/readable-stream/pull/30#issuecomment-2241086719).

```bash
npm i https://pkg.pr.new/@sec-ant/readable-stream@30
```

and change your code to this to check whether it fixes your problem?

```ts
for await (const event of asyncIterator(eventStream)) {
  ...
}
```

cc @ehmicky please be aware of this breaking change

**Comment by wong2** [comment]
Thanks, it works.

**Comment by Sec-ant** [comment]
This is released in v0.5.0
